thoth/messaging/message_base.py

```diff
@@ -31,6 +31,13 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+class BaseMessageContents(faust.Record, serializer="json"):  # type: ignore
+    """Default params for message contents."""
+
+    component_name: str = "WARNING: component_name not set"  # what component sent the message?
+    service_version: str = "WARNING: service_version not set"  # what version was that component?
```

Comment:
I don't know if this enforces devs of messages to provide such attributes. It looks to me that `faust.Record` uses attrs, so something like this should work:

```python
class BaseMessageContents(faust.Record, serializer="json"):  # type: ignore
    def __attrs_post_init__(self):
        if not getattr(self, "component_name") or not getattr(self, "service_version"):
           raise ...
        super().__attrs_post_init__()
```

Reply:
It doesn't, I did that on purpose so that it doesn't break current deployments. If we want to enforce devs to provide the attributes, all we need to do is not provide defaults.

Reply:
> If we want to enforce devs to provide the attributes, all we need to do is not provide defaults.

Should we patch all the components that use thoth-messaging to provide `component_name` and `service_version` with new thoth-messaging release?

Reply:
Yeah, and until the patch is put through both will just have warnings instead of the actual component and version info